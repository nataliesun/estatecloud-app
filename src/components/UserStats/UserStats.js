import React, { Component } from 'react';
import { Doughnut } from 'react-chartjs-2';
import DateApiService from '../../services/date-api-service';
import empty from './undraw_blank_canvas.svg';


import './UserStats.scss';
import DateContext from '../../contexts/DateContext';

class UserStats extends Component {
  static contextType = DateContext

  state = {
    reservations: {
      owned: null,
      made: null
    }
  }

  static defaultProps = {
    availability: [0, 0, 0],
    portfolio_value: 0
  }

  componentDidMount() {
    DateApiService.getReservationsForUser()
      .then(reservations => this.setState({ reservations }))
  }


  render() {
    const { availability, portfolio_value } = this.props;

    const showChart = availability.reduce((a, b) => a + b) >= 1 ? true : false;

    return (
      <div className="UserStats">
        <div className="stat-container">
          <div className="other-stats">
            {!showChart && <img src={empty} alt="blank_canvas" />}
            {showChart && <Doughnut
              data={createAvailabilityChart(availability)}
              options={{
                maintainAspectRatio: false,
                responsive: true,
                legend: {
                  display: false
                }
              }}
            />}
            <p>Availability</p>
          </div>
          <div className="other-stats">
            <h3>{this.state.reservations.owned}</h3>
            <p>Reservations at Owned Properties</p>
          </div>
          <div className="other-stats">
            <h3>{this.state.reservations.made}</h3>
            <p>Reservations Made</p>
          </div>
        </div>
        <div className="portfolio_value">
          <h3>{portfolio_value && portfolio_value.toLocaleString()}</h3>
          <p>Estimated Portfolio Value</p>
        </div>
      </div>
    );
  }
}

export default UserStats;

function createAvailabilityChart(availability) {
  const labels = ["Available", "Occupied", "Rented"]
  const data = availability


  return {
    labels,
    datasets: [
      {
        data,
        backgroundColor: [
          "rgb(255, 121, 121)", "#7979ff", "#70dfba"
        ]
      }
    ]
  }
}