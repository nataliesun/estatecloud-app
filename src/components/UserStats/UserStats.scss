@import '../variables.scss';

.UserStats {
  // border: 2px solid rgb(235, 235, 235);
  border-radius: 5px;
  margin-bottom: 20px;
  height: 30vh;
  display: flex;
  justify-content: space-around;
  flex-direction: column;
  box-shadow:  0 4px 8px 0 rgba(172, 172, 172, 0.2), 0 6px 20px 0 rgba(145, 145, 145, 0.19);
  // padding: 20px; 

  .stat-container {
    display: flex;
    flex-direction: row;
    flex: 3;
    border: 1px solid rgb(235, 235, 235);
    padding: 20px;
    width: 100%;
    height: 20vh;
  }

  .other-stats {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    // height: 60px;
    max-width: 50%;
    width: 50%;
    max-height: 20vh;
    &:first-of-type {
      border-right: 1px solid rgb(235, 235, 235);
    }

    p {
      padding-top: 10px;
      color: $main-text-color;
      opacity: .35;
    }
  }

  .portfolio_value {
    padding: 20px 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
    border-bottom: 1px solid rgb(235, 235, 235);
    border-left: 1px solid rgb(235, 235, 235);
    border-right: 1px solid rgb(235, 235, 235);
    width: 100%;
    flex: 1;
    
    p {
      color: $main-text-color;
      opacity: .35;
    }
  }
}