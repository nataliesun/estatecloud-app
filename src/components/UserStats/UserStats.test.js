import React from 'react';
import { shallow } from 'enzyme';
import UserStats from './UserStats';
import { library } from '@fortawesome/fontawesome-svg-core';
import 'jest-canvas-mock';

import {
  faDoorOpen, faPlusCircle, faPlus, faTimes
} from '@fortawesome/free-solid-svg-icons'

library.add(
  faDoorOpen, // logo
  faPlusCircle,
  faPlus,
  faTimes
)

describe(`UserStats Component`, () => {

  it('should render correctly with no props', () => {
    const component = shallow(<UserStats />);

    expect(component).toMatchSnapshot();
  });

  //write test that only calls createAvailabilityChart function using jest

})