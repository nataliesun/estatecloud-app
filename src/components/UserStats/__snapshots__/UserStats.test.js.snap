// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserStats Component should render correctly with no props 1`] = `
<div
  className="UserStats"
>
  <div
    className="stat-container"
  >
    <div
      className="other-stats"
    >
      <Doughnut
        data={
          Object {
            "datasets": Array [
              Object {
                "backgroundColor": Array [
                  "rgb(255, 121, 121)",
                  "#7979ff",
                  "#70dfba",
                ],
                "data": Array [
                  1,
                  0,
                  0,
                ],
              },
            ],
            "labels": Array [
              "Available",
              "Occupied",
              "Rented",
            ],
          }
        }
        options={
          Object {
            "legend": Object {
              "display": false,
            },
            "maintainAspectRatio": false,
            "responsive": true,
          }
        }
      />
      <p>
        Availability
      </p>
    </div>
    <div
      className="other-stats"
    >
      <h3 />
      <p>
        Reservations
      </p>
    </div>
  </div>
  <div
    className="portfolio_value"
  >
    <h3>
      0
    </h3>
    <p>
      Estimated Portfolio Value
    </p>
  </div>
</div>
`;
